tests/DataMapper/Product/Provider/ProductPricesProviderTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Tests\DataMapper\Product\Provider;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider\Exception\ProductPricesCannotBeProvidedException;
+use Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider\ProductPricesProvider;
+use Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider\ProductPricesProviderInterface;
+use Setono\SyliusMeilisearchPlugin\Tests\Application\Entity\Product;
+use Sylius\Component\Core\Model\Channel;
+use Sylius\Component\Core\Model\ChannelPricing;
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+
+final class ProductPricesProviderTest extends TestCase
+{
+    private MockObject&ProductVariantResolverInterface $productVariantResolver;
+
+    protected function setUp(): void
+    {
+        $this->productVariantResolver = $this->createMock(ProductVariantResolverInterface::class);
+    }
+
+    public function test_it_provides_prices_for_a_product(): void
+    {
+        $product = new Product();
+        $channel = new Channel();
+        $channel->setCode('MY_CHANNEL');
+
+        $channelPricing = new ChannelPricing();
+        $channelPricing->setChannelCode('MY_CHANNEL');
+        $channelPricing->setPrice(1000);
+        $channelPricing->setOriginalPrice(2000);
+
+        $productVariant = new ProductVariant();
+        $productVariant->addChannelPricing($channelPricing);
+
+        $this->productVariantResolver->method('getVariant')->willReturn($productVariant);
```

Comment:
If you want phpspec like prophecies, you can use this package: https://packagist.org/packages/phpspec/prophecy-phpunit

Reply:
I'll stick to PHPUnit Mock unless you decide we should standardize it and use prophecy :D